MergerService/Utils/TaskUtils.cs

```diff
@@ -1,46 +1,119 @@
+            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
+            string taskData = this._httpClient.PostDataString(url, null, false);
+
+            if (taskData is null)
+            {
+                return null;
+            }
+
+            try
+            {
+                var jsonSerializerSettings = new JsonSerializerSettings();
+                jsonSerializerSettings.Converters.Add(new StringEnumConverter());
+                return JsonConvert.DeserializeObject<MergeTask>(taskData, jsonSerializerSettings)!;
+            }
+            catch (Exception e)
+            {
+                this._logger.LogWarning(e, "Error serializing returned task");
+                return null;
+            }
+        }
+
+        public void NotifyOnCompletion(string jobId, string taskId)
+        {
+            // Notify overseer on task completion
+            string baseUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
+            string relativeUri = $"tasks/{jobId}/{taskId}/completed";
+            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
+            _ = this._httpClient.PostDataString(url, null, false);
+        }
+
+        private void Update(string jobId, string taskId, FormUrlEncodedContent content)
+        {
+            string baseUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
+            string relativeUri = $"jobs/{jobId}/tasks/{taskId}";
+            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
+            _ = this._httpClient.PutDataString(url, content, false);
+        }
+
+        public void UpdateProgress(string jobId, string taskId, int progress)
+        {
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("percentage", progress.ToString())
+            });
+            Update(jobId, taskId, content);
+        }
+
+        public void UpdateCompletion(string jobId, string taskId)
+        {
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("percentage", "100"),
+                new KeyValuePair<string, string>("status", "completed")
+            });
+            Update(jobId, taskId, content);
+        }
+
+        public void UpdateReject(string jobId, string taskId, int attempts, string reason, bool resettable)
+        {
+            attempts++;
+
+            // Check if the task should actually fail
+            if (!resettable || attempts == this._maxAttempts)
+            {
+                UpdateFailed(jobId, taskId, reason);
+            }
+
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("attempts", attempts.ToString()),
+                new KeyValuePair<string, string>("reason", reason),
+                new KeyValuePair<string, string>("resettable", resettable.ToString())
```

Comment:
i think you should also change the status to "PENDING"

Reply:
You are correct.